### PR TITLE
[DownloadBuildArtifactsV0] Add clean up logic of destination folder

### PR DESCRIPTION
**Pull request by tatyana-kostromskaya**
**Task name**: DownloadBuildArtifactsV0

**Description**: 
- Add input option "Clean destination folder" to advanced group
- Add logic to clean up destination folder in DownloadBuildArtifacts task if option "Clean destination folder" is checked

**Documentation changes required:** (Y/N) Y

**Added unit tests:** (Y/N) N

**Attached related issue:** https://github.com/microsoft/azure-pipelines-agent/issues/2711

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

Screenshot of added "checkbox":

image

### CODE REVIEW

**Review by max-zaytsev — APPROVED** [review]

**Review by DaniilShmelev — APPROVED** [review]
LGTM, thanks! Let me test this also

**Review by EzzhevNikita — APPROVED** [review]

File: Tasks/DownloadBuildArtifactsV0/main.ts [hunk 1]
```diff
@@ -109,6 +110,37 @@ async function main(): Promise<void> {
         });
         var artifacts = [];
 
+        // Clean destination folder if requested
+        if (cleanDestinationFolder) {
```
Comment (alexander-smolyakov) [inline Tasks/DownloadBuildArtifactsV0/main.ts:114]: Could it make sense to extract this logic into a function and move it to the `file_helper.ts` module?

File: Tasks/DownloadBuildArtifactsV0/main.ts [hunk 2]
```diff
@@ -109,6 +110,37 @@ async function main(): Promise<void> {
...
+            console.log(tl.loc('CleaningDestinationFolder', downloadPath));
+
+            // stat the destination folder (downloadPath) path
+            let destinationFolderStats: tl.FsStats;
+            try {
+                destinationFolderStats = tl.stats(downloadPath);
+            }
+            catch (err) {
```
Comment (alexander-smolyakov) [inline Tasks/DownloadBuildArtifactsV0/main.ts:122]:
```suggestion
          } catch (err) {
```
Could you please fix the code formatting?

File: Tasks/DownloadBuildArtifactsV0/main.ts (hunk 1 above)
Comment (tatyana-kostromskaya) [reply Tasks/DownloadBuildArtifactsV0/main.ts:114]: Resolved

File: Tasks/DownloadBuildArtifactsV0/main.ts [hunk 3]
```diff
@@ -109,6 +110,37 @@ async function main(): Promise<void> {
...
+                if (err.code != 'ENOENT') {
```
[the remark this answers is not in the document]
Comment (tatyana-kostromskaya) [reply Tasks/DownloadBuildArtifactsV0/main.ts]: Resolved

File: Tasks/DownloadBuildArtifactsV0/main.ts (hunk 2 above)
Comment (tatyana-kostromskaya) [reply Tasks/DownloadBuildArtifactsV0/main.ts]: Resolved

**Review by alexander-smolyakov — APPROVED** [review]
LGTM, thanks!

**Review by anatolybolshakov — APPROVED** [review]

**Pull request merged**